web/client_service_indexeddb.ts

```diff
@@ -0,0 +1,111 @@
+import Dexie, {Table} from 'dexie';
+import {KEYUTIL, b64utohex} from 'jsrsasign';
+import {Properties, CLIENT_PROPERTIES_FIELD} from './common/client_config';
+import {ClientService} from './client_service';
+import {isNonEmptyString, isInteger} from './common/polyfill/is';
+
+const KEYSTORE_DATABASE_NAME = 'scalar';
+
+class ScalarIndexedDb extends Dexie {
+  keystore!: Table<{id: string; key: CryptoKey}, string>;
+  certstore!: Table<{id: string; cert: string}, string>;
+
+  constructor() {
+    super(KEYSTORE_DATABASE_NAME);
+    this.version(1).stores({keystore: 'id', certstore: 'id'});
+  }
+}
+
+export class ClientServiceWithIndexedDb extends ClientService {
+  static async create(properties: Properties) {
+    const holderId = properties[CLIENT_PROPERTIES_FIELD.CERT_HOLDER_ID];
+    const certVersion = properties[CLIENT_PROPERTIES_FIELD.CERT_VERSION];
+
+    if (!isNonEmptyString(holderId) || !isInteger(certVersion)) {
+      throw new Error(
+        `${CLIENT_PROPERTIES_FIELD.CERT_HOLDER_ID} (string) and ${CLIENT_PROPERTIES_FIELD.CERT_VERSION} (integer) are required`
+      );
+    }
+
+    const cryptoKey = properties[
+      CLIENT_PROPERTIES_FIELD.PRIVATE_KEY_CRYPTOKEY
+    ] as CryptoKey;
+    const keyPem = properties[
+      CLIENT_PROPERTIES_FIELD.PRIVATE_KEY_PEM
+    ] as string;
+    const certPem = properties[CLIENT_PROPERTIES_FIELD.CERT_PEM] as string;
+    const id = `${holderId}_${certVersion}`;
+
+    const db = new ScalarIndexedDb();
+
+    let key: CryptoKey;
+    if (cryptoKey || keyPem) {
+      key = cryptoKey || (await toCryptoKeyFromJwk(toJwkFromPkcs1(keyPem)));
+      await db.keystore.put({id: id, key: key});
+    } else {
+      const item = await db.keystore.get(id);
+
+      if (!item) {
+        throw new Error('Can not find key from indexedDB');
```

Comment:
Not sure the official name, but should "i" be capitalized?

```suggestion
        throw new Error('Can not find key from IndexedDB');
```

Reply:
How about this? It's super minor, but PTAL if you get a chance.